createmsi.py

```diff
@@ -46,23 +46,34 @@ def __init__(self, jsonfile):
         self.update_guid = jsondata['update_guid']
         self.basename = jsondata['name_base']
         self.need_msvcrt = jsondata.get('need_msvcrt', False)
+        self.arch = jsondata.get('arch', 0)
         self.main_xml = self.basename + '.wxs'
         self.main_o = self.basename + '.wixobj'
         self.bytesize = 32 if '32' in platform.architecture()[0] else 64
         # rely on the environment variable since python architecture may not be the same as system architecture
         if 'PROGRAMFILES(X86)' in os.environ:
             self.bytesize = 64
```

Comment:
FWICT this fails on native Windows with 32 bits because `self.bytesize` won't get set. The logic should probably go something like:

```
if arch exists
  self.bytesize = arch
else
  if programfiles(x86)
    self.bytesize = 64
  else
    self.bytesize = 32